3rdparty/Makefile

```diff
@@ -1,3 +1,14 @@
+#
+# Note that this Makefile is borked and it will rebuild everything every time you run it. This is due
+# me depending on $(BUILD_DIR) everywhere.
```

Comment:
It is easy to fix by dropping `mkdir -p $(BUILD_DIR)` in every target instead of depending on `$(BUILD_DIR)`.